axelrod/fingerprint.py

```diff
@@ -0,0 +1,294 @@
+    probes are created using the `JossAnnTransformer`.
+
+    Parameters
+    ----------
+    probe : class
+        A class that must be descended from axelrod.strategies.
+    coordinates : list of tuples
+        Tuples of length 2 representing each coordinate, eg. (x, y)
+
+    Returns
+    ----------
+    probe_dict : ordered dictionary
+        An Ordered Dictionary where the keys are tuples representing each
+        coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+        parameters that correspond to (x, y).
+    """
+    probes = [create_jossann(coordinate, probe) for coordinate in coordinates]
+    return probes
+
+
+def create_edges(coordinates):
+    """Creates a set of edges for a spatial tournament.
+
+    Constructs edges that correspond to `coordinates`. Coordinates that sum to
+    1 or less will have edges that start at 0, those who sum to more than will
+    have an edge that starts at 1.
+
+    Parameters
+    ----------
+    coordinates : list of tuples
+        Tuples of length 2 representing each coordinate, eg. (x, y)
+
+    Returns
+    ----------
+    edges : list of tuples
+        A list containing tuples of length 2. All tuples will have either 0
+        or 1 as the first element. The second element is the index of the
+        corresponding probe (+2 to allow for including the Strategy and it's
+        Dual).
+    """
+    edges = []
+    for index, coordinate in enumerate(coordinates):
+        #  Add 2 to the index because we will have to allow for the Strategy
+        #  and it's Dual
+        if sum(coordinate) > 1:
+            edge = (1, index + 2)
+        else:
+            edge = (0, index + 2)
+        edges.append(edge)
+    return edges
+
+
+def generate_data(interactions, coordinates, edges):
+    """Generates useful data from a spatial tournament.
+
+    Matches interactions from `results` to their corresponding coordinate in
+    `probe_coords`.
+
+    Parameters
+    ----------
+    interactions : dictionary
+        A dictionary of the interactions of a tournament
+    coordinates : list of tuples
+        A list of tuples of length 2, where each tuple represents a
+        coordinate, eg. (x, y).
+    edges : list of tuples
+        A list containing tuples of length 2. All tuples will have either 0
+        or 1 as the first element. The second element is the index of the
+        corresponding probe (+2 to allow for including the Strategy and it's
+        Dual).
+
+    Returns
+    ----------
+    coordinate_scores : dictionary
+        A dictionary where the keys are coordinates of the form (x, y) and
+        the values are the mean score for the corresponding interactions.
+    """
+    edge_scores = [np.mean([cfspt(scores) for scores in interactions[edge]])
+                   for edge in edges]
+    coordinate_scores = dict(zip(coordinates, edge_scores))
+    return coordinate_scores
+
+
+def read_interactions(filename):
```

Comment:
This exists in `interactions_utils`? If you're doing something slightly different here I suggest you simply modify the output from what you would get from `interactions_utils`, (there is no need to duplicate the code for reading in the file).